src/NUnitFramework/nunitlite.tests/TextUIReportTests.cs

```diff
@@ -120,6 +120,8 @@ public void SummaryReportTest()
                 "    Duration: 0.123 seconds\n\n"));
         }
 
+        // Requires mock-assembly to be compiled with /optimize-
+        // or else the test methods do tail calls and the stack frame disappears
```

Comment:
Are you sure about the tail calls? I would expect it to be method inlining and not tail calls (not that it matters much 😄 )

Reply:
Definitely not inlining. The test method is invoked directly via reflection so inlining is impossible, and it's the test method that is the missing stack frame. If Assert.Warn inlines into the test method that's totally fine because we still see the test method stack frame, but if the test method tail-calls Assert.Warn then the test method stack frame is lost before the call.